test/unit/mwApiCapabilities.test.ts

```diff
@@ -41,30 +58,48 @@ describe('Checking Mediawiki capabilities', () => {
     expect(await MediaWiki.hasWikimediaMobileApi()).toBe(false)
     expect(await MediaWiki.hasRestApi()).toBe(true)
     expect(await MediaWiki.hasVisualEditorApi()).toBe(true)
+    expect(await MediaWiki.checkIndexPhpPath()).toBe(true)
+    expect(await MediaWiki.checkWikiPath()).toBe(true)
   })
 
   test('test capabilities of minecraft.wiki with correct VisualEditor receipt', async () => {
     MediaWiki.base = 'https://minecraft.wiki'
-    MediaWiki.wikiPath = '/'
+    MediaWiki.wikiPath = '/w/'
     MediaWiki.actionApiPath = '/api.php'
+    MediaWiki.indexPhpPath = '/index.php'
     Downloader.init = { uaString: `${config.userAgent} (contact@kiwix.org)`, speed: 1, reqTimeout: 1000 * 60, webp: true, optimisationCacheUrl: '' }
 
     expect(await MediaWiki.hasWikimediaDesktopApi()).toBe(false)
     expect(await MediaWiki.hasWikimediaMobileApi()).toBe(false)
     expect(await MediaWiki.hasRestApi()).toBe(false)
     expect(await MediaWiki.hasVisualEditorApi()).toBe(true)
+    expect(await MediaWiki.checkIndexPhpPath()).toBe(true)
+    expect(await MediaWiki.checkWikiPath()).toBe(true)
+    expect(await MediaWiki.checkIndexPhpPath('MCE')).toBe(true) // check with a redirected article
+    expect(await MediaWiki.checkWikiPath('MCE')).toBe(true) // check with a redirected article
+  })
+
+  test('test capabilities of minecraft.wiki with incorrect path config', async () => {
+    MediaWiki.base = 'https://minecraft.wiki'
+    Downloader.init = { uaString: `${config.userAgent} (contact@kiwix.org)`, speed: 1, reqTimeout: 1000 * 60, webp: true, optimisationCacheUrl: '' }
+    expect(await MediaWiki.checkIndexPhpPath()).toBe(false)
+    expect(await MediaWiki.checkWikiPath()).toBe(true) // automatically redirect by minecraft.wiki
```

Comment:
This means we actually failed to confirm the validity of mwWikiPath. Our check says `/wiki/` is the correct path for minecraft.wiki but all wiki links use `/w/` so we still end up with a ZIM where all links got removed.

Reply:
The only redirects we want to allow are page redirects, so we should confirm that our mwWikiPath is still part of the url after the redirect.

Though as noted in https://github.com/openzim/mwoffliner/issues/2421#issuecomment-3058687417 I'm not sure these checks are actually needed as #2422 would remove both of these CLI arguments.

Reply:
I'm pretty sure we will have wiki in the wild which will redirect from say `/wiki/` to `/index.php?title=...`, so we must accept redirects on wiki path as well.

I don't know how to have a bullet-proof solution, and a good solution is anyway a mix of https://github.com/openzim/mwoffliner/issues/2421 (this PR) and https://github.com/openzim/mwoffliner/issues/2422

Reply:
> Though as noted in https://github.com/openzim/mwoffliner/issues/2421#issuecomment-3058687417 I'm not sure these checks are actually needed as https://github.com/openzim/mwoffliner/issues/2422 would remove both of these CLI arguments.

This is indeed a good point, I started this PR before you mentioned in #2422 that we have a reliable way to retrieve these values and drop the CLI arguments. Maybe we should just focus on #2422 and drop this PR. Tbh I'm a bit concerned about the fact that the logic of #2422 would fail under some condition and we would miss the CLI arguments.